pkg/apiserver/rest/dualwriter.go

```diff
@@ -149,16 +148,10 @@ type ServerLockService interface {
 func SetDualWritingMode(
 	ctx context.Context,
 	kvs NamespacedKVStore,
-	legacy LegacyStorage,
-	storage Storage,
-	entity string,
-	desiredMode DualWriterMode,
-	reg prometheus.Registerer,
-	serverLockService ServerLockService,
-	requestInfo *request.RequestInfo,
+	cfg *SyncerConfig,
```

Comment:
seems this should be required.  does it need to be a pointer?  passing nil will cause a panic

i think its a pointer because it gets mutated.  maybe we should return a new config rather than mutating?

Reply:
We usually pass configs as pointer to avoid copying too many values. The nil check is a good shout, will add it.